### ISSUE TITLE
Parser ignores "."

### ISSUE THREAD

**Issue by mossprescott** [slamdata/scala-pathy#23]
PathCodec's parse methods ignore directories called `.`, which is really a posix-specific feature and doesn't make sense for other codecs. 

Much worse, parsing a file called `.` will actually produce a path with a non-sensical type:

```scala
@ posixCodec.parseAbsFile("/foo/bar/.").map(posixCodec.unsafePrintPath)
res1: Option[String] = Some(/foo/bar/)
```

**Comment by mossprescott** [comment]
After further consideration, I don't think the PathCodec abstraction is even usable for URI paths (the intended use) because it's not clear what should happen if you try to print a path that contains `ParentIn` or `Current` if the codec has no representation analogous to `..` and `.`. Unless we think it makes sense to make `printPath` partial, we'll probably have to roll our own parsing for URIs in quasar.

**Comment by mossprescott** [comment]
@jdesah: just noticed this sitting in my review queue and figured I'd ping you and see if you're still planning to take it over.

**Comment by jedesah** [comment]
Fixed by #27
